src/editor.rs

```diff
@@ -626,6 +642,9 @@ impl Editor {
                 prompt_mode = Some(PromptMode::Find(String::new(), self.cursor.clone(), None)),
             Key::Char(GOTO) => prompt_mode = Some(PromptMode::GoTo(String::new())),
             Key::Char(DUPLICATE) => self.duplicate_current_row(),
+            Key::Char(CUT) => self.copy_current_row(false),
```

Comment:
What do you think of removing the `is_preserving` argument for `copy_current_row`, and instead replacing this line with

```rust
Key::Char(CUT) => {
    self.copy_current_row();
    self.delete_current_row();
}
```

IMO it is slightly clearer if `copy_current_row` only does the copy, and it doesn't affect the line count.